vermouth/processors/apply_rubber_band.py

```diff
@@ -309,10 +322,14 @@ def apply_rubber_band(molecule, selector,
     constants = compute_force_constants(distance_matrix, lower_bound,
                                         upper_bound, decay_factor, decay_power,
                                         base_constant, minimum_force)
+    # we select the nodes here so the selection list has the same
+    # order in build_connectivity_matrix and build_pair matrix
+    selected_nodes = _apply_selection(molecule, selection=selection)
+    print(selected_nodes)
     connected = build_connectivity_matrix(molecule, res_min_dist - 1,
-                                          selection=selection)
+                                          selected_nodes=selected_nodes)
     same_domain = build_pair_matrix(molecule, domain_criterion,
-                                    selection=selection)
+                                    selected_nodes=selected_nodes)
```

Comment:
Since selected_nodes is used multiple times, it should definitely be a list or tuple. i.e. not a generator.

Reply:
done